### PR TITLE
Remove Request concept

### PR DESCRIPTION
**Pull request by fcsonline**
In order to simplify the api of `mobx-rest`, we are removing the
`Request` wrapper. Now, `Collection` and `Model` will return raw
`Promise`

### CODE REVIEW

File: src/Collection.ts [hunk 1]
```diff
@@ -334,19 +332,10 @@ export default abstract class Collection<T extends Model> extends Base {
   create(
     attributesOrModel: { [key: string]: any } | T,
     { optimistic = true, path }: CreateOptions = {}
-  ): Request {
+  ): Promise<any> {
```
Comment (masylum) [inline src/Collection.ts:335]: we can type that any

File: src/Collection.ts [hunk 2]
```diff
@@ -356,15 +345,27 @@ export default abstract class Collection<T extends Model> extends Base {
    * use the options to disable adding, changing
    * or removing.
    */
-  fetch({ data, ...otherOptions }: SetOptions = {}): Request {
-    const { abort, promise } = apiClient().get(this.url(), data, otherOptions)
```
Comment (masylum) [inline src/Collection.ts:360]: we lose abort capabilities (which people were not using unluckily), although this has changed a lot nowadays with the introduction to https://developer.mozilla.org/en-US/docs/Web/API/AbortController

**Review by masylum — COMMENTED** [review]
This looks fantastic! The API is much cleaner and I'm sure this is going to work better with promise rejection errors and sentry.

File: src/Collection.ts (hunk 1 above)
Comment (fcsonline) [reply src/Collection.ts:335]: With which type?

Comment (fcsonline) [reply src/Collection.ts:335]: `Promise<T>`?

File: README.md [hunk 3]
```diff
@@ -740,7 +679,7 @@ Developed and battle tested in production in [Factorial](https://factorialhr.com
 
 (The MIT License)
 
-Copyright (c) 2019 Pau Ramon <masylum@gmail.com>
+Copyright (c) 2022 Pau Ramon <masylum@gmail.com>
```
Comment (masylum) [inline README.md:682]: 🤦

File: src/Collection.ts [hunk 4]
```diff
@@ -356,15 +346,31 @@ export default abstract class Collection<T extends Model> extends Base {
...
+  fetch({ data, ...otherOptions }: SetOptions = {}): Promise<any> {
+    const { promise } = apiClient().get(this.url(), data, otherOptions)
 
-    promise
+    return promise
```
Comment (masylum) [inline src/Collection.ts:352]: we can now use `await + try/catch` instead of `then + catch`

File: README.md (hunk 3 above)
Comment (fcsonline) [reply README.md:682]: jejejej

File: src/Collection.ts [hunk 5]
```diff
@@ -356,15 +346,31 @@ export default abstract class Collection<T extends Model> extends Base {
...
       .then(data => {
         if (Array.isArray(data)) this.set(data, otherOptions)
+
+        return data
+      })
+      .catch(error => {
+        throw new ErrorObject(error)
```
[the remark this answers is not in the document]
Comment (fcsonline) [reply src/Collection.ts:359]: Because this was performed by `withRequest`. I move it back here and model too

File: src/Collection.ts (hunk 4 above)
Comment (fcsonline) [reply src/Collection.ts:352]: In a another PR

**Pull request merged**